src/gui/DatabaseTabWidget.cpp

```diff
@@ -353,80 +361,78 @@ bool DatabaseTabWidget::saveDatabase(Database* db)
 
 bool DatabaseTabWidget::saveDatabaseAs(Database* db)
 {
-    DatabaseManagerStruct& dbStruct = m_dbList[db];
-    QString oldFileName;
-    if (dbStruct.saveToFilename) {
-        oldFileName = dbStruct.filePath;
-    }
-    else {
-        oldFileName = tr("New database").append(".kdbx");
-    }
-    QString fileName = fileDialog()->getSaveFileName(this, tr("Save database as"),
-                                                     oldFileName, tr("KeePass 2 Database").append(" (*.kdbx)"),
-                                                     nullptr, 0, "kdbx");
-    if (!fileName.isEmpty()) {
-        QFileInfo fileInfo(fileName);
-        QString lockFilePath;
-        if (fileInfo.exists()) {
-            // returns empty string when file doesn't exist
-            lockFilePath = fileInfo.canonicalPath();
-        }
-        else {
-            lockFilePath = fileInfo.absolutePath();
+    while (true) {
```

Comment:
How would you solve it? return saveDatabaseAs(db) instead of continue only gobbles up the stack unnecessarily.
As an alternative approach I could move all the code into a separate method and then only put the method call into an endless loop.